### PR TITLE
docs: update helm documentation

### PR DESCRIPTION
**Pull request by anthonycorletti**
### SUMMARY
Noticed https://github.com/apache/superset/issues/11363 and updated documentation to reflect a success full deployment on GKE with helm3.

```sh
$ helm version && echo && kubectl version && echo && kubectl get po -A
version.BuildInfo{Version:"v3.5.1", GitCommit:"32c22239423b3b4ba6706d450bd044baffdcf9e6", GitTreeState:"dirty", GoVersion:"go1.15.7"}

Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-14T12:50:19Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"17+", GitVersion:"v1.17.14-gke.1600", GitCommit:"7c407f5cc8632f9af5a2657f220963aa7f1c46e7", GitTreeState:"clean", BuildDate:"2020-12-07T09:22:27Z", GoVersion:"go1.13.15b4", Compiler:"gc", Platform:"linux/amd64"}

NAMESPACE     NAME                                                        READY   STATUS      RESTARTS   AGE
default       superset-77f5b77848-cchs2                                   1/1     Running     0          32m
default       superset-init-db-8q6xw                                      0/1     Completed   0          32m
default       superset-postgresql-0                                       1/1     Running     0          32m
default       superset-redis-master-0                                     1/1     Running     0          32m
default       superset-worker-86945968b4-56zdf                            1/1     Running     0          32m
kube-system   event-exporter-gke-666b7ffbf7-sv5wg                         2/2     Running     0          93m
kube-system   fluentbit-gke-42qh8                                         2/2     Running     0          93m
kube-system   fluentbit-gke-gq544                                         2/2     Running     0          93m
kube-system   fluentbit-gke-zjtd7                                         2/2     Running     0          93m
kube-system   gke-metrics-agent-2qs5z                                     1/1     Running     0          93m
kube-system   gke-metrics-agent-6n7bm                                     1/1     Running     0          93m
kube-system   gke-metrics-agent-fjfvk                                     1/1     Running     0          93m
kube-system   kube-dns-9c59558bb-wvmkf                                    4/4     Running     0          93m
kube-system   kube-dns-9c59558bb-xrqnk                                    4/4     Running     0          93m
kube-system   kube-dns-autoscaler-5c78d65cd9-q4ng2                        1/1     Running     0          93m
kube-system   kube-proxy-gke-superset-test-default-pool-900766ff-1cwt     1/1     Running     0          93m
kube-system   kube-proxy-gke-superset-test-default-pool-900766ff-k1nt     1/1     Running     0          93m
kube-system   kube-proxy-gke-superset-test-default-pool-900766ff-vvj4     1/1     Running     0          93m
kube-system   l7-default-backend-5b76b455d-db4kl                          1/1     Running     0          93m
kube-system   metrics-server-v0.3.6-547dc87f5f-44kbj                      2/2     Running     0          92m
kube-system   prometheus-to-sd-7rrkt                                      1/1     Running     0          93m
kube-system   prometheus-to-sd-fxvxk                                      1/1     Running     0          93m
kube-system   prometheus-to-sd-thc28                                      1/1     Running     0          93m
kube-system   stackdriver-metadata-agent-cluster-level-56d688675d-lsdqr   2/2     Running     0          92m
```

### ADDITIONAL INFORMATION
Can publish these on merge wrt https://github.com/apache/superset/blob/master/docs/README.md#to-publish

### CODE REVIEW

**Comment by anthcor** [comment]
noting that `requirements.yaml` is deprecated in helm 3 https://helm.sh/docs/faq/#consolidation-of-requirementsyaml-into-chartyaml

**Comment by amitmiran137** [comment]
Look like a re doing of #11763

File: helm/superset/Chart.yaml [hunk 1]
```diff
@@ -23,3 +23,12 @@ maintainers:
     email: cychiang0823@gmail.com
     url: https://github.com/cychiang
 version: 0.1.0
+dependencies:
```
Comment (craig-rueda) [inline helm/superset/Chart.yaml:26]: The API version needs to bump to v2 for requirements to live here.

**Review by craig-rueda — CHANGES_REQUESTED** [review]
Left inline comment. Please also remove requiments.yaml / requirements.locl in favor of keeping everything in Chart.yaml.

**Review by amitmiran137 — APPROVED** [review]

**Review by craig-rueda — APPROVED** [review]

**Pull request merged**